src/main/resources/db/changelog/changes/09-create-books-categories-table.yaml

```diff
@@ -0,0 +1,38 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-books-categories-table
+      author: Mykhailo Kornukh
+      changes:
+        - createTable:
+            tableName: books_categories
+            columns:
+              - column:
+                  name: book_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: category_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+        - createIndex:
+            indexName: idx_books_categories_book
```

Comment:
You should remove both create index statements because DB generates indexes for PK and FK by default. So just add FK constraint to columns